dashboard/lib/services/user/password_resetter.rb

```diff
@@ -0,0 +1,58 @@
+module Services
+  module User
+    class PasswordResetter < Services::Base
+      attr_reader :email, :user
+
+      def initialize(email:)
+        @email = email
+        @user = nil
+      end
+
+      def call
+        if email.blank?
+          @user = ::User.new
+          user.errors.add :email, ::I18n.t('activerecord.errors.messages.blank')
+          return user
+        end
+        # We are no longer sending an email to parents, so grab the first user we find
+        # (a user with an Email auth option first, otherwise any user that has that email)
+        @user = ::User.find_by_email_or_hashed_email(email)
+        if user.nil?
+          not_found_user = ::User.new(email: email)
+          Cdo::Metrics.put(
+            'User', 'PasswordResetUserNotFound', 1, {
+              Environment: CDO.rack_env
+            }
+          )
+          return not_found_user
+        end
```

Comment:
```suggestion
      attr_reader :email

      def initialize(email:)
        @email = email
      end
      
      private def user
        return @user if defined? @user

        if email.blank?
          @user = ::User.new
          @user.errors.add :email, ::I18n.t('activerecord.errors.messages.blank')
        else
          # We are no longer sending an email to parents, so grab the first user we find
          # (a user with an Email auth option first, otherwise any user that has that email)
          @user = ::User.find_by_email_or_hashed_email(email)
        end

        if @user.nil?
          @user = ::User.new(email: email)
          @user.errors.add :email, :not_found
          Cdo::Metrics.put(
            'User', 'PasswordResetUserNotFound', 1, {
              Environment: CDO.rack_env
            }
          )
        end
        
        @user
      end

      def call
        return user if user.errors.present?

```

Reply:
We are not displaying an error if the user is not found and we ended up combining metrics, so I took out the second half of this method, but I did take the rest of this suggestion. Thank you!

Reply:
Do you mean this is no longer relevant?
https://github.com/code-dot-org/code-dot-org/pull/61691/files#diff-0a5e79b422e8d2a2b8e5d9cc2203f62899ee74b65da749306b3d5f7c8b21a2d2L1715-L1719

Reply:
Do you mean this is no longer relevant?
https://github.com/code-dot-org/code-dot-org/pull/61691/files#diff-0a5e79b422e8d2a2b8e5d9cc2203f62899ee74b65da749306b3d5f7c8b21a2d2L1727-L1731